geode-docs/developing/query_select/aggregates.html.md.erb

```diff
@@ -0,0 +1,340 @@
+---
+
+<!--
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+The aggregate functions 
+`MIN`, `MAX`, 
+`COUNT`, `COUNT` over a `DISTINCT` expression, 
+`SUM`, `SUM` over a `DISTINCT` expression, 
+`AVG`, `AVG` over a `DISTINCT` expression, and 
+`GROUP BY` are supported.
+
+The following sections provide descriptions and limitations (if any) about the aggregate functions, how do they work and how they can be used. 
+
+## <a id="groupBy"></a>GROUP BY
+You can collect data across multiple entries and group the results by one or more columns through the usage of the `GROUP BY` statement.
+It's important to note some facts about its usage:
+
+* It can group results by one or more fields.
+* It returns a single record / entry per group.
+* It must always be placed before the `ORDER BY` clause, if any.
+* It can be used in conjunction with other aggregate functions: `MIN`, `MAX`, `COUNT`, `SUM` and `AVG`.
+* It groups records using the selected fields if and only if the fields have identical data across entries.
+* It is required, whenever an aggregate function is used within a query with other selected fields, to also use `GROUP BY`.
+* If there are no other aggregate functions within the query, all fields included within a GROUP BY clause must also be part of the original projection list, and all fields included within the projection list must also be part of the `GROUP BY` clause.
+
+The following are example `GROUP BY` queries.
+
+The following `GROUP BY` query returns the maximum amount of sales per employee.
+
+``` pre
+SELECT ID, MAX(e.sales)
+FROM /employees e
+GROUP BY ID
+```
+
+The following `GROUP BY` query returns the minimum, maximum, total count, average and summation of IDs grouped by status.
+
+``` pre
+SELECT pf.status, MIN(pf.ID), MAX(pf.ID), COUNT(pf.ID), AVG(pf.ID), SUM(pf.ID)
+FROM /portfolio pf
+GROUP BY pf.status
+```
+
+## <a id="min"></a>MIN
+
+The `MIN` keyword returns the minimum or smallest value from the selected expression. The expression itself must always evaluate to `java.lang.Comparable`. The `MIN` statement returns the actual type of the selected element as its result. 
+
+The following are example `MIN` queries that return region entries (the entries implement the `java.lang.Comparable` interface).
```

Comment:
Personally, I think this example should go last of the MIN examples.  I think the one with min(pf.ID) is easiest to understand then then after seeing that, this one makes more sense?.  Just personal preference I think

Reply:
Thanks for reviewing @jhuynh1.
I can change this if you have strong opinions about it, but all the aggregate functions start with examples that show how to return the region entry directly (it's not uncommon to try to find the minimum/maximum entry within a region). In order to keep the documentation about aggregate function consistent with the rest, I included these as the first examples for `MIN` and `MAX`.
Anyway, let me know if you want it changed.

Reply:
I'm ok, I just thought finding a min entry is interesting in that, how is in determined but I guess it's the same for an ID

Reply:
Sure, thanks for the feedback.
Long story short: it doesn't matter what we use as the argument for `MIN` / `MAX` **as long as it implements the `java.lang.Comparable` interface**, that's what we use to calculate the final result:

```

  @Override
  public void accumulate(Object value) {
    if (value == null || value == QueryService.UNDEFINED) {
      return;
    }

    Comparable comparable = (Comparable) value;

    if (currentOptima == null) {
      currentOptima = comparable;
    } else {
      @SuppressWarnings("unchecked")
      int compare = currentOptima.compareTo(comparable);

      if (findMax) {
        currentOptima = compare < 0 ? comparable : currentOptima;
      } else {
        currentOptima = compare > 0 ? comparable : currentOptima;
      }
    }
  }

```

That requirement is already part of the improved docs so I'll merge this PR :-).
If you think there's anything missing, please let me know, I'll be happy to open a new ticket and improve the docs further!.
Cheers.